src/framework/script/script-type.js

```diff
@@ -234,6 +257,67 @@ class ScriptType extends EventHandler {
         return this._enabled && !this._destroyed && this.entity.script.enabled && this.entity.enabled;
     }
 
+    /**
+     * Listen for an event on an EventHandler. The ScriptType will add and remove the event listeners on the
+     * EventHandler automatically when the ScriptType is enabled, disabled and destroyed.
+     *
+     * @param {EventHandler} eventHandler - EventHandler to listen for events on.
+     * @param {string} name - Name of the event to bind the callback to.
+     * @param {HandleEventCallback} callback - Function that is called when event is fired. Note
+     * the callback is limited to 8 arguments.
+     * @param {object} [scope] - Object to use as 'this' when the event is fired.
+     * @returns {ScriptType} Self for chaining.
+     * @example
+     * this.listen(this.app.mouse, 'mousemove', this.onMouseMove, this);
+     * @see {@link ScriptType#unlisten} to remove listeners to an event on an EventHandler.
+     */
+    listen(eventHandler, name, callback, scope) {
+        this._listeners.push({
+            eventHandler,
+            name,
+            callback,
+            scope
+        });
+
+        if (this.enabled) {
+            eventHandler.on(name, callback, scope);
+        }
+
+        return this;
+    }
+
+    /**
+     * Removes a listener for an event on an EventHandler that was added by {@link ScriptType#listen}.
+     *
+     * @param {EventHandler} eventHandler - EventHandler that was originally listened to.
+     * @param {string} name - Name of the event that was originally listened to.
+     * @param {HandleEventCallback} callback - Function that was used as the callback when the event was originally
+     * listened to.
+     * @param {object} [scope] - Object that was used as the scope when the event was originally listened to
+     * @returns {ScriptType} Self for chaining.
+     * @example
+     * this.unlisten(this.app.mouse, 'mousemove', this.onMouseMove, this);
+     * @see {@link ScriptType#listen} to listen to events on an EventHandler.
+     */
+    unlisten(eventHandler, name, callback, scope) {
```

Comment:
This a slow function but I don't expect this to be called very often/at all and the number of subscribed events is generally expected to be quite low generally.